File: Day 3/Task 2/Task_2.py
def ItemPriority(item):
    if (item.isupper()):
        return ord(item) - ord('A') + 27

    return ord(item) - ord('a') + 1


fhandle = open("day_3_1.txt")

prioritySum = 0
backpacks = list()
numLinesRead = 0

for line in fhandle:
    backpacks.append(line.rstrip())
    numLinesRead = numLinesRead + 1
    if numLinesRead < 3:
        continue

    numLinesRead = 0

    for i in range(len(backpacks[0])):
        if backpacks[1].find(backpacks[0][i]) != -1 and backpacks[2].find(backpacks[0][i]) != -1:
            prioritySum += ItemPriority(backpacks[0][i])
            break
    backpacks = list()



print(prioritySum)

File: Day 3/Task 2/Task 2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>
#include <vector>


int BadgePriority(char badge) {
	if (std::isupper(badge))
		return badge - 'A' + 27;
	return badge - 'a' + 1;
}

int main()
{
	std::ifstream fhandle("day_3_1.txt");
	int prioritySum = 0;
	while (true) {

		std::vector<std::string> backpacks(3);
		//read three inputs
		std::getline(fhandle, backpacks.at(0));
		std::getline(fhandle, backpacks.at(1));
		std::getline(fhandle, backpacks.at(2));

		if (!fhandle)
			break;

		bool badgeFound = false;

		for(int i = 0; i < backpacks[0].size() && !badgeFound; i++)
			for (int j = 0; j < backpacks[1].size() && !badgeFound; j++)
				if (backpacks[0].at(i) == backpacks[1].at(j)) {

					auto it = std::find(std::begin(backpacks[2]), std::end(backpacks[2]), backpacks[1].at(j));

					if (it != std::end(backpacks[2])) {
						badgeFound = true;
						prioritySum += BadgePriority(*it);
					}
				}
				
	}
	std::cout << "The sum of priorities of all badges of each three-Elf group is: " << prioritySum;

}

